core/src/main/java/io/grpc/internal/GzipInflatingBuffer.java

```diff
@@ -230,6 +236,16 @@ int getAndResetBytesConsumed() {
   }
 
   /**
+   * Reports bytes consumed by the inflater since the last invocation of this method, then resets
+   * the count to zero.
+   */
+  int getAndResetDeflatedBytesConsumed() {
```

Comment:
This is used for the "wire size"? It seems fair to include the gzip overhead as part of the wire size (i.e., use `getAndResetBytesConsumed()`). We're doing that with per-message compression today.

I'm okay with excluding the gzip overhead (like this method does), but it looks like it just adds Yet More Stuff to track. So I'd rather just keep it simple.

Reply:
I spoke with @zhangkun83 offline about this. Currently [MessageDeframer](https://github.com/grpc/grpc-java/blob/7e534ed704441cdef7f2af6c094148acf3dabd8b/core/src/main/java/io/grpc/internal/MessageDeframer.java#L292) only reports gRPC body bytes as the wire size, ignoring gRPC message header bytes. Separating out the gzip overhead makes it possible to continue doing this - otherwise we will sometimes get gzip bytes included in the counts and sometimes not, depending on the message deframer state when the gzip metadata gets processed.